src/components/AttachmentModal.js

```diff
@@ -74,52 +140,17 @@ class AttachmentModal extends Component {
             <>
                 <ModalWithHeader
                     type={CONST.MODAL.MODAL_TYPE.CENTERED}
-                    onClose={() => this.setState({isModalOpen: false})}
-                    isVisible={this.state.isModalOpen}
-                    title={this.props.title}
+                    onClose={() => clearAttachmentModalData()}
+                    isVisible={this.props.visibleAttachment.isModalOpen}
+                    title={this.state.title}
                     backgroundColor={themeColors.componentBG}
                 >
                     <View style={attachmentViewStyles}>
-                        {this.state.sourceURL && (
-                            <AttachmentView sourceURL={sourceURL} file={this.state.file} />
+                        {this.props.visibleAttachment.sourceURL && (
+                            <AttachmentView sourceURL={sourceURL} file={this.props.visibleAttachment.file} />
```

Comment:
A lot of these edits to change props to state can be avoided if we just pass the "active attachment" down from the parent instead of subscribing to updates here.